raiden/blockchain/utils.py

```diff
@@ -0,0 +1,91 @@
+import math
+
+from eth_typing import BlockNumber
+from structlog import get_logger
+
+from raiden.blockchain.exceptions import BlockBatchSizeTooSmall
+from raiden.settings import BlockBatchSizeConfig
+
+log = get_logger(__name__)
+
+
+class BlockBatchSizeAdjuster:
+    """ Helper to dynamically adjust the block batch size.
+
+    Internally it uses an exponential function of base ``base`` onto which the block range given
+    in the config is mapped.
+
+    The default values for ``base`` and ``step_size`` fit well for ranges that span several orders
+    of magnitude. For very small ranges those values may need to be adjusted.
+    """
+
+    def __init__(
+        self,
+        block_batch_size_config: BlockBatchSizeConfig,
+        base: float = 2.0,
+        step_size: float = 1.0,
+    ) -> None:
+        self._block_batch_size_config = block_batch_size_config
+        self._base = base
+        self._step_size = step_size
+
+        self._scale_max = math.log(self._block_batch_size_config.max, self._base) + 1
```

Comment:
@ulope What are those used for?

Reply:
Nothing at all. Both are left over from a needlessly more complicated version I had before.

Nice catch, thanks.